pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -706,6 +725,7 @@ func (ku *Upgrader) copyCustomNodeProperties(client armhelpers.KubernetesClient,
 		}
 	}
 
+	newNode.Spec.Unschedulable = false
```

Comment:
should this be done after the node is updated?

Reply:
Based on the tests for the edge case that this PR tries to fix, it seems ok to put "copying node properties and re-enabling scheduling' in the same UpdateNode() call.

Reply:
If it's done after the update then the pending pods will be incorrectly scheduled on this node before it receives the ported taints. Unless you see a better way to do it

Reply:
Isn't it the opposite? We want to make the node schedulable _after_ it has the correct taints no? otherwise pods might get scheduled incorrectly in the short time before it gets the ported taints.

Reply:
Correct, so making the node schedulable after all the taints are applied only. 
I believe we're saying the same thing. So will stand down for Charlie to explain the actual code.

Reply:
Although using single UpdateNode() call to apply node properties and make it schedulable works fine according to tests.  I change the code to use two UpdateNode() calls separately to apply the node properties first and make the node schedulable next.